sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -831,6 +838,13 @@ class Analyzer(
             val withName = Alias(agg, s"_w${extractedExprBuffer.length}")()
             extractedExprBuffer += withName
             withName.toAttribute
+
+          case ne: Alias if hasWindowFunction(ne) && !hasAggregation(ne) =>
+            ne.children.map(_.transform {
+              case e: NamedExpression => extractExpr(e)
+            })
+            ne
+
```

Comment:
A one-line change can fix this and pass your test: `case attr: Attribute => extractExpr(attr)`

I think the missing case is `Attribute`, not `Alias`.